kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/predicates/In.java

```diff
@@ -116,8 +116,11 @@ public String toString() {
   }
 
   public String toString(Schema schema) {
-    return Joiner.on(',').join(
-        Iterables.transform(set, new ToString<T>(schema)));
+    // we want the string this returns to be consistent
+    // ensure the set is sorted before calling toString on it.
+
+    Iterable<String> strings = Iterables.transform(set, new ToString<T>(schema));
+    return Joiner.on(',').join(Sets.newTreeSet(strings));
```

Comment:
The parsing for In that uses a HashSet rather than a LinkedHashSet is up at [line 45](https://github.com/kite-sdk/kite/pull/348/files#diff-b835a100cf3c183c62f4afa63550e5e1R45). I can open up a new issue for that.

I think it's a stretch here to say that this code path is only when needed. This change would be hit on the normal call of View.toURI() and would result in a normalized URI. I can look at a tweak where the URI is normalized only when the caller requests it.